### PR TITLE
Update to 3.18.4

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit <SvenDowideit@docker.com>

### CODE REVIEW

**Comment by SvenDowideit** [comment]
@tianon can we play with overlayfs?

File: Dockerfile [hunk 1]
```diff
@@ -18,9 +18,9 @@ RUN apt-get update && apt-get -y install  unzip \
                         pkg-config \
                         p7zip-full
 
-ENV KERNEL_VERSION  3.16.7
-ENV AUFS_BRANCH     aufs3.16
-ENV AUFS_COMMIT     b3883c3cb86937801fdd2e188032063de617ef73
+ENV KERNEL_VERSION  3.18.1
+ENV AUFS_BRANCH     aufs3.18
```
Comment (tianon) [inline Dockerfile]: Reading mailing lists upstream, it looks like they switched to the `aufs3.18.1+` branch for the 3.18 development.  Does it make sense for us to switch too?

**Comment by tianon** [comment]
I'm definitely +1 on playing with `overlay` and thus updating to 3.18+.  Can we get the `Dockerfile`+`kernel_config` changes in a PR by themselves so we can discuss the other stuff separately?

File: Dockerfile (hunk 1 above)
Comment (SvenDowideit) [inline Dockerfile]: I presumed that `aufs3.18+` will contain changes that would apply to eventually going to be `3.18.2` - whereas this branch would be specific to `3.18.1`

**Comment by SvenDowideit** [comment]
the kernel_config changes are needed for the new kernel to compile - just doing the usual `make oldconfig` resulted in a broken kernel.

yes, I can take the Dockerfile and Makefile out, but I'm not sure that gains much.

**Comment by clkao** [comment]
When can we get this in?  Can't wait to play with overlay

**Comment by tianon** [comment]
> Can we get the `Dockerfile`+`kernel_config` changes in a PR by themselves

I'm not asking for the `Dockerfile` to be separated from the `kernel_config`; that would indeed be ludicrous and gain nothing.  I just want the `Makefile` separately because it warrants more discussion than a kernel bump.

**Comment by SvenDowideit** [comment]
mmm, I'll run a test with 3.18.4 too
 @tianon ?

**Comment by SvenDowideit** [comment]
https://github.com/boot2docker/boot2docker/issues/704 is a kernel crash in 3.16.7, which is so totally EOL, and its replacement 3.17 is also EOL.

@tianon @bfirsh

**Comment by tianon** [comment]
Yeah, this is really late in the 1.5.0 game, but I think it's worth squeezing this into rc3 and thus release.

LGTM

**Comment by tianon** [comment]
(if anyone disagrees with us fixing some bugs and getting newer shinies, now's the time to speak up :smile:)

**Pull request merged**